File: Yandex/Practicum/Algorightms_and_Data_Structures/7_sprint/A._Биржа/solution.cpp
#include <iostream>
#include <vector>


void solution() {
    /*
    Time Complexity O(N)
    Memory Complexity O(1)
    */
    int n;
    std::cin >> n;
    std::vector<int> nums (n, 0);
    for (int i = 0; i < n; ++i) {
        std::cin >> nums[i];
    }
    int profit = 0;
    int curPrice = 0;
    for (int idx = 0; idx < n; ++idx) {
        if (!idx || nums[idx] < curPrice) {
            curPrice = nums[idx];
        } else {
            profit += nums[idx] - curPrice;
            curPrice = nums[idx];
        }
    }
    std::cout << profit << "\n";
}


int main() {
    solution();
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/7_sprint/A._Биржа/solution.py
import sys


def solution() -> None:
    """
    Time Complexity O(N)
    Memory Complexity O(1)
    """
    n: int = int(sys.stdin.readline().rstrip())
    nums: list[int] = list(map(int, sys.stdin.readline().rstrip().split()))
    profit: int = 0
    cur_price: int = 0
    for idx in range(n):
        if not idx or nums[idx] < cur_price:
            cur_price = nums[idx]
        else:
            profit += nums[idx] - cur_price
            cur_price = nums[idx]
    sys.stdout.write(f'{profit}\n')


if __name__ == '__main__':
    solution()